proposals/NNNN-vehicle-type-filter.md

```diff
@@ -0,0 +1,435 @@
+# Enable OEM exclusive apps support
+
+* Proposal: [SDL-NNNN](NNNN-vehicle-type-filter.md)
+* Author: [Ashwin Karemore](https://github.com/ashwink11)
+* Status: **Under review**
+* Impacted Platforms: [Core / iOS / Java Suite / Protocol]
+
+## Introduction
+
+This feature will enable SDL adapters to provide exclusive apps to their users depending on vehicle type. The proposal describes a way to share vehicle type information before sending the Register App Interface request.
+
+## Motivation
+
+The proposal tries to mitigate the below issues.
+
+### Handling Proprietary app registrations
+
+In the current implementation of the SDL, vehicle information is shared with the `Register app interface` response. If there are any apps, that work with specific OEMs, they would need to send the `Register app interface` RPC to know vehicle details. If the vehicle details do not match with the supported vehicle types, they would need to implement some mechanism to unregister the app. This behavior would cause an app to be shown on SDL enabled system for some time until the app unregisters itself from the IVI system. The users would see that apps were shown in the IVI system for a moment and are removed within seconds. By providing vehicle type information before app registration we could solve this issue, the app will not register on IVI if vehicle type is not supported, allowing app partners and SDL adapters to provide exclusive apps experience to their users depending on vehicle type information. 
```

Comment:
```suggestion
In the current implementation of SDL, vehicle information is shared with the `RegisterAppInterface` response. If there are any apps that work with specific OEMs, they would need to send the `RegisterAppInterface` RPC to know vehicle details. If the vehicle details do not match the supported vehicle types, they would need to implement some mechanism to unregister the app. This behavior would cause an app to be shown on an SDL enabled system for some time until the app unregisters itself from the IVI system. The users would see that apps were shown in the IVI system for a moment, but are removed within seconds. By providing vehicle type information before app registration, we could solve this issue. The app will not register on the IVI if vehicle type is not supported, allowing app partners and SDL adopters to provide an exclusive apps experience to their users depending on vehicle type information. 
```